nibe/connection/nibegw.py

```diff
@@ -537,7 +537,7 @@ def _encode(self, obj, context, path):
         )
     ),
     "checksum" / Checksum(Int8ub, xor8, this.fields.data),
-).compile()
+)
```

Comment:
This is strange. But for some reason the code crashes when streams end with some internal errors instead of handling the Terminated() construct.

Reply:
Ok. Checking terminated first does the trick.